File: 17502.cpp
#include <iostream>
using namespace std;

int main() {
	int N; cin >> N;
	string pal; cin >> pal;

	for (int i = 0; i <= (N - 1) / 2; i++) {
		int right_index = N - i - 1;
		if (pal[i] == '?' && pal[right_index] == '?')
			pal[i] = pal[right_index] = 'a';
		else if (pal[i] == '?')
			pal[i] = pal[right_index];
		else
			pal[right_index] = pal[i];
	}

	cout << pal << endl;
}

File: 17502.py
N = int(input())
pal = input()
pals = [pal[i] for i in range(len(pal))]

for i in range((N - 1) // 2 + 1):
    right_index = N - i - 1
    if pals[i] == '?' and pals[right_index] == '?':
        pals[i] = pals[right_index] = 'a'
    elif pals[i] == '?':
        pals[i] = pals[right_index]
    else:
        pals[right_index] = pals[i]

for p in pals:
    print(p, end='')
print()
